converter.go

```diff
@@ -427,11 +427,7 @@ func arrowToValue(
 	case realType:
 		for i, flt64 := range array.NewFloat64Data(data).Float64Values() {
 			if !srcValue.IsNull(i) {
-				if higherPrecision {
```

Comment:
This check shouldn't be used for native Go types, only high precision arrow types that require big.* structs.